lib/realtime/tenants/authorization.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Realtime.Tenants.Authorization do
+  alias Plug.Conn
+  alias Phoenix.Socket
+
+  require Logger
+
+  def get_authorizations(%Socket{} = socket, db_conn, params) do
+    with {:ok, {:ok, permissions}} <- get_permissions_for_connection(db_conn, params) do
+      {:ok, Socket.assign(socket, :permissions, permissions)}
+    else
+      {:error, _} -> {:error, :unauthorized}
+    end
+  end
+
+  def get_authorizations(%Conn{} = conn, db_conn, params) do
+    with {:ok, {:ok, permissions}} <- get_permissions_for_connection(db_conn, params) do
+      {:ok, Conn.assign(conn, :permissions, permissions)}
+    else
+      {:error, _} -> {:error, :unauthorized}
+    end
+  end
+
+  defp get_permissions_for_connection(conn, params) do
```

Comment:
```suggestion
```suggestion
  @spec get_permissions_for_connection(DBConnection.conn(), map()) ::
          {:ok, map()} | {:error, any()}
  defp get_permissions_for_connection(conn, params) do
```
  defp get_permissions_for_connection(conn, params) do
```